tooling-docker/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -77,7 +82,42 @@ protected void start() throws Exception {
             }
         });
 
-        containerReadyCondition.waitUntilReady(containerReadyTimeout);
+        long startTime = System.currentTimeMillis();
+        while (!isContainerReady()) {
+            if (System.currentTimeMillis() - startTime > containerReadyTimeout) {
+                throw new DockerTimeoutException("Container was not ready in timeout : " + containerReadyTimeout + " ms");
+            }
+            // fail fast mechanism in case of malformed docker command, for example bad arguments, invalid format of port mapping, image version,...
+            if (!dockerRunProcess.isAlive() && dockerRunProcess.exitValue() != 0) {
+                throw new DockerException("Starting of docker container using command: \"" + String.join(" ", cmd)
+                        + "\" failed. Check that provided command is correct.");
+            }
+        }
+    }
+
+    private boolean isContainerReady() throws Exception {
+        CompletableFuture<Boolean> condition = new CompletableFuture().supplyAsync(() -> containerReadyCondition.isReady());
+        try {
+            return condition.get(containerReadyTimeout, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException ex) {
+            // in case condition hangs interrupt it so there are no zombie threads
+            condition.cancel(true);
+
+            throw new ContainerReadyConditionException("Provided ContainerReadyCondition.isReady() method took longer than containerReadyTimeout: "
+                    + containerReadyTimeout + " ms. Check it does not hang and does not take longer then containerReadyTimeout. " +
+                    "It's expected that ContainerReadyCondition.isReady() method is short lived (takes less than 1 second).", ex);
+        }
+    }
+
+    private void checkDockerPresent() throws Exception {
+        Process dockerRunProcess = new ProcessBuilder()
+                .redirectErrorStream(true)
+                .command(new String[] {"docker", "info"})
+                .start();
+        dockerRunProcess.waitFor();
+        if (dockerRunProcess.exitValue() != 0)   {
+            throw new DockerException("Docker is not present on this machine. It must be installed for running tests with docker container.");
```

Comment:
I'd rephrase 
`Docker is not present on this machine. It must be installed for running tests with docker container.`
to
`Docker is either not present or not installed on this machine. It must be installed and started up for executing tests with docker container.`

Reply:
thanks, fixed